### ISSUE TITLE
An error is reported on bb.edn :enter hook

### ISSUE THREAD

**Issue by lread** [clj-kondo/clj-kondo#1584]
**version**

```shell
$ clj-kondo --version                                                                                                                                                                           
clj-kondo v2022.02.09
```

**platform**

macOS 10.15.7

**problem**

Some forms of the `:enter` hook are generating lint errors.

**repro**

Given a minimal `bb.edn` in a new empty directory:

```Clojure
{:tasks {:enter (let [name (:name (current-task))] (println "Task enter" name))
         test (println "test")}}
```

If I run: 
```Shell
clj-kondo --lint bb.edn
```

**expected behavior**

I expect to see no errors reports and a 0 exit code.

**actual behavior**
I do get a 0 exit code, but I see the following output:
```
ERROR:  No value supplied for key: (println "Task enter" name)
linting took 7ms, errors: 0, warnings: 0
```

**observations**
If I remove all keywords from `:enter` hook form...
```Clojure
{:tasks {:enter (println "boo")
         test (println "test")}}
```

...then behavior matches expectations.

Could be I am currently not seeing a misuse/abuse? Dunno.

**Comment by mknoszlig** [comment]
Thanks for catching this! Cycle detection needed to be a bit more selective. I'm wondering, shouldn't tests fail on something like this? there's this `try`/`catch` clause in `deps_edn.clj` that just prints the error. should it crash at least optionally?

**Comment by borkdude** [comment]
The try/catch is there to not crash clj-kondo when you call `sexpr` which is a non-safe operation.

**Comment by lread** [comment]
Thanks for the fix!

I suppose clj-kondo could collect these unexpected throws to make them easier to discover during test runs? (If that's what you were kind of getting at @mknoszlig?)

**Comment by mknoszlig** [comment]
@lread yes, that's the direction i was going for. if i'd had a test case that covered your case above, there would have still been a chance we would've missed it because the tests would've passed. while it makes sense that clj-kondo shouldn't crash there, it would certeinly help (imho) to have some convenient way of check nothing went wrong during testing.
